### PR TITLE
Fix Coverity NullReferenceException Issues w/ Remotes

### PR DESCRIPTION
**Pull request by whoisj**
The issue here is that the method asserts that both parameters are null or neither are, then proceeds to behave like only one is. This results in a NRE too deep into the stack to easily debug/resolve the issue. Moving the error upstream makes it easier to identify the problem.

Question: should we just throw at the call-site if both arguments (`remote` and `url`) are `null`?

Included some beautification as a separate commit.

### CODE REVIEW

File: LibGit2Sharp/Network.cs [hunk 1]
```diff
@@ -111,13 +111,13 @@ static RemoteSafeHandle BuildRemoteSafeHandle(RepositorySafeHandle repoHandle, R
...
 
-            RemoteSafeHandle remoteHandle;
+            RemoteSafeHandle remoteHandle = null;
 
             if (url != null)
             {
                 remoteHandle = Proxy.git_remote_create_anonymous(repoHandle, url);
             }
-            else
+            else if (remote != null)
```
Comment (nulltoken) [inline LibGit2Sharp/Network.cs]: We're actually asserting to protect against this. And the call to `Debug.Assert` is "seen" by Coverity, however it looks like the tool doesn't properly take it into account in the code flow analysis.

Comment (jamill) [inline LibGit2Sharp/Network.cs]: yeah - the ` ^ ` operator is the exclusive-or operator.

Comment (whoisj) [inline LibGit2Sharp/Network.cs]: Coverity is correct. We in fact do assert but only in debug mode. We then have code flow for when both are null which should be an impossibility. We even dereference a null value in the malformed code flow that we assert, but do not check, will not happen.

Comment (jamill) [inline LibGit2Sharp/Network.cs]: this is correct. However, this is not a public method so we do control all callers of this method. Either way, I agree that this is a bit fragile. What would you think about throwing an exception in the case where we get unexpected input? Or, performing the refactoring suggested in #1084. As you mentioned, I think everyone agrees on the ultimate structure for this code (two methods).

**Comment by whoisj** [comment]
Superceded by #1087

**Pull request closed without merging**